### PR TITLE
Fix/258 download button

### PR DESCRIPTION
**Pull request by Jeremy1026**
Created a new partial (download/_button) to manage a download button that automatically selects the proper OS. Applied this partial on the home page and on /get. If no download is available for the current OS the button on the home page links to /get which allows users to enter their email address to receive notifications of new releases.

### CODE REVIEW

File: controller/action/DownloadActions.class.php [hunk 1]
```diff
@@ -28,6 +28,29 @@ public static function executeGetDaemonRedirect(string $os)
...
+  }
+
+  public static function prepareButtonPartial() {
+    if (static::prepareDownloadInformation()) {
+      return static::prepareDownloadInformation();
+    }
+    return [];
+  }
+
+  private static function prepareDownloadInformation() {
```
Comment (kauffj) [inline controller/action/DownloadActions.class.php]: `protected` over `private` generally. But in the case of `prepareXXX` functions, they should be `public`.

`prepareFoo` will be called automatically when including `_foo.php` as a view.

**Comment by Jeremy1026** [comment]
Changes made per request @kauffj

**Comment by kauffj** [comment]
@Jeremy1026 can you rebase on master? Looks like PR is bringing in unrelated commits.

**Comment by tzarebczan** [comment]
@Jeremy1026 you may want to open a new PR for this unless you can re-base just your changes somehow. Looks like this one got pretty messy. Thanks!

**Comment by tzarebczan** [comment]
@Jeremy1026 not sure if you tried to fix something (got a github notification) but it still looks messy.

**Comment by tzarebczan** [comment]
@Jeremy1026 - closing this, but please take another shot when you are available :)

**Pull request closed without merging**